### PR TITLE
Replace UNIX sockets with TCP sockets in Ray on Windows

### PR DESCRIPTION
**Pull request by mehrdadn**
## Why are these changes needed?

Windows compatibility

## Related issue number

#631

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

**Comment by robertnishihara** [comment]
@mehrdadn do you expect any performance implications here? We should test performance microbenchmarks on this before merging.

File: src/ray/util/url.cc [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+#include <algorithm>
+#include <string>
+
+#include <boost/asio/ip/address.hpp>
+#include <boost/asio/ip/tcp.hpp>
+
+#include "ray/util/logging.h"
+#include "ray/util/url.h"
+
+boost::asio::ip::tcp::endpoint parse_ip_tcp_endpoint(
```
Comment (robertnishihara) [inline src/ray/util/url.cc]: This function should be documented. Not sure what it does.

**Comment by mehrdadn** [comment]
@robertnishihara sorry if my title was misleading, I'm only replacing UNIX sockets with TCP sockets on Windows, so I don't expect any performance implications on other platforms. On Windows yes this will be slower than more lightweight features like named pipes, but those are considerably more involved.

File: src/ray/util/url.cc (hunk 1 above)
Comment (mehrdadn) [reply src/ray/util/url.cc]: It just [parses the host & port of a URL](/ray-project/ray/pull/6823/files/d8ad4c4a1569c264c82226eb19a0794298089bc1#diff-a67b85512722e126b455ba21a0c99349R15). I added a comment to the header.

**Review by pcmoritz — APPROVED** [review]

**Pull request merged**